src/controllers/controller.doughnut.js

```diff
@@ -262,6 +263,20 @@ module.exports = function(Chart) {
 			} else {
 				return 0;
 			}
-		}
+		},
+		
+		//gets the max border or hover width to properly scale pie charts
+        getMaxBorderWidth: function (elements) {
+            var max = 0;
+
+            for (var i = 0; i < elements.length; i++) {
+                var borderWidth = elements[i]._model ? elements[i]._model.borderWidth : 0,
```

Comment:
I would cache `elements.length`, `this.index`, the current element and move declarations outside the loop:

    var max = 0;
    var index = this.index;
    var element, borderWidth, hoverWidth;
    
    for (var i = 0, ilen = elements.length; i < ilen; ++i) {
         element = elements[i];
         borderWidth = element._model ? element._model.borderWidth : 0,
         hoverWidth = elements._chart ? element._chart.config.data.datasets[index].hoverBorderWidth : 0;
    }